src/cargo/sources/registry/mod.rs

```diff
@@ -496,6 +496,46 @@ impl<'cfg> RegistrySource<'cfg> {
                 )
             }
 
+            // We need to check wether the any of the path items contains
+            // a reserved name in Windows.
+            #[cfg(target_os = "windows")]
+            {
+                use crate::util::restricted_names::is_windows_reserved;
+                let mut split_path: Vec<&str> = entry_path
+                    .as_os_str()
+                    .to_str()
+                    // If the path contains non UTF-8 characters, provide the
+                    // propper error.
+                    .ok_or(crate::util::errors::ProcessError {
+                        desc: String::from("Invalid UTF8 character in the file path"),
+                        exit: None,
+                        output: None,
+                    })
+                    .chain_err(|| "Invalid UTF8 character in the file path")?
```

Comment:
I don't know if that's needed, but It would be weird if the path contains a non-UTF-8 character and the process just panics pointing to the `unwrap()` that would replace this code. 

If you feel that this should not be checked here, I can just `unwrap()` and that's it. (Or if you have better suggestions, happy to hear & apply them!)